### ISSUE TITLE
What about to Donate the SourceCode to Apache NetBeans

### ISSUE THREAD

**Issue by gilbertoca** [madflow/flow-netbeans-markdown#128]
Dear @madflow and @stengerh  ,

This plugin is really popular among NetBeans users and I would like to ask you the possibility of the source code donation to Apache NetBeans, so it can be an integral part of all Netbeans ecosystem?
WTA?

ref: https://github.com/apache/netbeans/pull/1428

**Comment by madflow** [comment]
@gilbertoca @jlahoda I personally would embrace the idea that Markdown would become a 1st class citizen in Netbeans. Most of the source code has been written by @stengerh and @junichi11 . The MIT License would allow a transition without asking every contributor I would assume. I would still like to here some opinions on this (but, everyone could just adopt/fork it without asking permissions).

A blocker is this issue in my opinion: https://github.com/madflow/flow-netbeans-markdown/issues/121

Pegdown was the perfect fit back in the days - now the plugin will need a Markdown parser update to perform at its best.

**Comment by jlahoda** [comment]
@matthiasblaesing, would you have any insights on what could be a way to get this plugin to be part of NetBeans itself?

Regarding the parser, I'll try to find some to look at the parsers, but cannot make promises.

**Comment by Chris2011** [comment]
Please also have a look at this ticket #121. It's about using other parsers than the deprecated pegdown.

**Comment by junichi11** [comment]
@madflow I respect your opinion because this is your repo ;)

**Comment by matthiasblaesing** [comment]
This is my understanding of the Apache Legal FAQ: We can include MIT licensed code into the netbeans code base. When we do this, we still have to follow the requirements of the license. We would therefor be required to include the MIT copyright information. We have a similar case in the smooth scrolling code in the "Look & Feel Customization Library":

https://github.com/apache/netbeans/blob/master/platform/o.n.swing.plaf/licenseinfo.xml#L47..L56

here the ALv2 is worse than the MIT license. For the MIT license, I would reference the license and be done with it, the ALv2 requires us to clutter the NOTICE file with this.

My take on this:
- if it is possible to gather all people that contributed to the project and let them transfer their work/copyright to a single author, who then donates it to the ASF would be the best (yeah, probably won't happen)
- alternative 1: Get all contributors to agree on a relicensing to ALv2 _without_ a NOTICE file (have a look at our NOTICE file or the lucene one and you know, that they are a _very_ bad idea)
- alternative 2: The code gets contributed as is and the license file of netbeans gets one more entry

**Comment by madflow** [comment]
I am really not familiar how to "contribute/donate" source code to bigger project like Netbeans. I have never heard of a `NOTICE` file before. I am open to "asking" all 8 Contributors if they would object relicensing if this would make things easier.  I would favor `alternative 2`.

**Comment by stengerh** [comment]
Sorry for the late response. I was swamped with work for the last months and must have missed the notification.

I would agree on a relicensing. Alternative 2 would be fine for me as well if someone else objects to a relicensing.

**Comment by Chris2011** [comment]
So do we still need to donate the code? Afaik, @jlahoda already started it: https://github.com/apache/netbeans/pull/1428 also he started to add flexmark. So I think the code for donation is incompatible?

**Comment by jlahoda** [comment]
My patch was for just highlighting  - the donation would give us more fully featured support. I think the more fully featured support would be much better - and my PR could just be closed as not needed anymore.

**Comment by Chris2011** [comment]
Ok, thx for letting me know. And to add this here, donating the code will still use the old and deprecated pegdown lib. Just to let you know.

**Comment by ve3** [comment]
This plugin is not working anymore in Netbeans 12.1.

**Comment by thomas-mc-work** [comment]
On my machine it's still working with NetBeans 12.1. Did you try a fresh install, @ve3 ?

**Comment by ve3** [comment]
> 
> 
> On my machine it's still working with NetBeans 12.1. Did you try a fresh install, @ve3 ?

Yes, I tried that but when it was installed and restart the plugin did not appear in the list and not functional.

I'm using Windows 10 x64 Netbeans 12.1.

**Comment by agarcialeon** [comment]
I can confirm that 2.3.2 version works on NetBeans 12.3 (Preview works, navigator and syntax highlighting).
Anyway, it's anyone working on an update to flexmark? Is there any ETA?

Because my company has switched to Netbeans as a publisher, I have made a list of plugins, themes and other resources so that other developers can more easily find the links after seeing that the plugins portal is down.

**Comment by Chris2011** [comment]
There was another Markdown plugin: https://github.com/jkost/MarkdownViewerForNetBeans. @jkost is on it, but has some trouble. What is your list, that you mentioned?

**Comment by agarcialeon** [comment]
Is this one: [Awesome Netbeans](https://github.com/agarcialeon/awesome-netbeans).
I'm using the "awesome" kind of repo for this since I haven't found any other with enough contents and I already have other lists to help me gather resources.

**Comment by Chris2011** [comment]
Oh that's great, thx :). We should tweet about it and send them around. This is a great resource.

**Comment by Chris2011** [comment]
BTW: https://github.com/apache/netbeans/pull/1428 there is now basic syntax highlighting coming in NetBeans 12.4 added by @jlahoda. Just syntax highlighting. Converting to see the result is still missing and can be handled in an other ticket with flexmark. e.g.

**Comment by agarcialeon** [comment]
Ah, good to see it! Will check it when I have more free time. Thanks for the info.

**Comment by ve3** [comment]
I cannot even install.
Download from this page https://github.com/madflow/flow-netbeans-markdown/releases/tag/2.3.2 ( flow-netbeans-markdown.nbm )

Try to install with Netbeans 12.3 but error.

> The validation of downloaded plugins cannot be completed, cause: NBM ...\flow-netbeans-markdown.nbm needs unpack200 to process following entries:
> ...

**Comment by Chris2011** [comment]
@ve3 that is because of your JDK that you use. From one specific JDK version, I think >= JDK 14, they removed the unpack200 package. So workaround is to use another JDK < 14 to run it with NetBeans and install the plugin and then switch back if really needed. This happens, for some plugins that have packed stuff within.

**Comment by agarcialeon** [comment]
> 
> 
> @ve3 that is because of your JDK that you use. From one specific JDK version, I think >= JDK 14, they removed the unpack200 package. So workaround is to use another JDK < 14 to run it with NetBeans and install the plugin and then switch back if really needed. This happens, for some plugins that have packed stuff within.

I link to [stackoverflow](https://stackoverflow.com/questions/61626523/apache-netbeans-11-3-c-plugin-timeout-unable-to-install) with the same response as @Chris2011. 
For reference, @ve3 I'm using OpenJDK 11.0.10.9.

**Comment by jkost** [comment]
Hallo guys,

nice page. I think you need to mention the list of NetBeans plugins
<https://cwiki.apache.org/confluence/display/NETBEANS/Community+plugins>,
which is disappointing, w.r.t. to the plugins that existed in versions 8.2
or earlier, as well as the Pro Apache NetBeans
<https://www.amazon.com/Pro-Apache-NetBeans-Building-Applications/dp/1484253698>
book but also many other books on NetBeans
<https://netbeans.apache.org/kb/docs/platform.html#_netbeans_platform_books>
(I 'll send you the list).

However, I 'd rather add all this information in the Apache NetBeans web
site, instead of blogs here and there.

Regarding the markdown plugin, I didn't have time to work on it, and I hope
to find some from next week on. I owe it to Chris. :)

Have a nice weekend.

John.

**Comment by agarcialeon** [comment]
The book you refer is already added. But I can link the other ones.
About the plugins list, I wasn't aware of it. Definitely will add it too. But yes, I can contribute to the website with the content listed there.

**Comment by agarcialeon** [comment]
@jkost do you have updated links for the blogs? I have already added the books to the list.

**Comment by jkost** [comment]
Hallo Alvaro,

below a list of books on NetBeans. I have created two categories, one about
NetBeans IDE and another one about the NetBeans Platform (RCP) and I would
advise you do the same:

   - IDE
      - Myatt, A. (2007), *Pro NetBeans IDE 5.5 Enterprise Edition*, APress.
      - Salter D. & Jennings F. (2008), *Building SOA-Based Composite
      Applications Using NetBeans IDE 6*, Packt.
      - Heffelfinger D.R. (2008), *Java EE 5 Development with NetBeans 6*,
      Packt.
      - Kutler C. & Leonard B. (2009), *Netbeans Ruby and Rails IDE with
      JRuby*, APress.
      - Heffelfinger D.R. (2011), *Java EE 6 Development with NetBeans 7*,
      Packt.
      - Dantas R. (2011), *NetBeans IDE 7 Cookbook*, Packt.
      - Wielenga G. (2015), *Beginning NetBeans IDE*, APress.
      - Heffelfinger D.R. (2015), *Java EE 7 Development with NetBeans 8*,
      Packt.
      - Salter D. (2015), *Mastering NetBeans*, Packt.
      - Salter D. & Dantas R. (2014), *NetBeans IDE 8 Cookbook*, Packt.
      - Oracle (2014), *Developing Applications with NetBeans IDE*, Release
      8.0, E50452-06.
      - Oracle (2016), *Developing Applications with NetBeans IDE*, Release
      8.1, E61618-03.
      - Bourdeau T. et. al. (2002), *NetBeans: The Definitive Guide*,
      O'Reilly.
      - *Champenois L. et. al. (2005), NetBeans IDE Field Guide: Developing
      Desktop, Web, Enterprise, And Mobile Applications
      <https://www.amazon.com/NetBeans%C2%BF-IDE-Field-Guide-Applications/dp/0131876201/ref=sr_1_1?dchild=1&keywords=NetBeans+field+guide&qid=1620156409&s=books&sr=1-1>.*
      - Palandurkar* A. (2013), Instant NetBeans IDE How-to*, Packt.
      - RCP
      - Myatt A. (2008), *Pro Netbeans IDE 6 Rich Client Platform Edition*,
      APress.
      - Anderson P. & Anderson G. (2015), *JavaFX for NetBeans Platform*,
      Addison-Wesley.
      - Wexbridge J. & Nyland W. (2014), *NetBeans Platform for Beginners*,
      Leanpub <https://leanpub.com/nbp4beginners>.
   - Tulach J. (2008), *Practical API Design,* APress.

Kind regards,

John.

**Comment by jkost** [comment]
You may also add my blog <https://nbrcp.wordpress.com/>, as well as another
blog <https://netbeansrcp.wordpress.com/> I came across.

I do have there links to other blogs, too.

John.

**Comment by thomas-mc-work** [comment]
While I like your engagement around Netbeans, I'd prefer to keep the comments in this issues aligned to the topic. I think it would be better to place these hints in [the issues sections of that respective repository](https://github.com/agarcialeon/awesome-netbeans/issues).

Thanks!

**Comment by Chris2011** [comment]
In the discussions section would be better. FYI.

**Comment by agarcialeon** [comment]
I agree with @thomas-mc-work and I have added @jkost references to the list and opened threads in the discussions section of the repository to not pollute this repository issue. Thanks again @Chris2011 for the tip.

**Comment by jkost** [comment]
Ideally I would like to do pull requests but we started this thread so I
continued on that

Στις Τετ, 5 Μαΐ 2021, 10:14 π.μ. ο χρήστης Álvaro García León <
***@***.***> έγραψε:

**Comment by geertjanw** [comment]
There's interest in continuing to develop this plugin and also the source code would be welcome at Apache NetBeans to be continued there: https://github.com/apache/netbeans/discussions/3477
